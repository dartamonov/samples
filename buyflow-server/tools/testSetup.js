// Tests are placed alongside files under test.
// This file does the following:
// 1. Sets the environment to 'production' so that
//    dev-specific babel config in .babelrc doesn't run.
// 2. Disables Webpack-specific features that Mocha doesn't understand.
// 3. Registers babel for transpiling our code for testing.

// This assures the .babelrc dev config (which includes
// hot module reloading code) doesn't apply for tests.
// process.env.NODE_ENV = 'production';
// process.env.NODE_ENV = 'test';

// Register babel so that it will transpile ES6 to ES5
// before our tests run.
require('babel-register')();
//require("babel-polyfill");
require("babel-jest");
