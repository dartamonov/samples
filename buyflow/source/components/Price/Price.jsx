import React, { Component, PropTypes } from 'react';
import _ from 'lodash';

class SimplePrice extends Component {

  render() {
    const {
      price,
    } = this.props;

    //if (_.isEmpty(price)) return null;

    const dollarsCents = (price > 0) ? _.words(price.toString(10), /[^.]+/g) : ['0', '00'];
    const dollars = dollarsCents[0];
    const cents = _.size(dollarsCents[1]) === 1 ? `.${dollarsCents[1]}0` : `.${dollarsCents[1]}`.substring(0, 3);

    return (
      <span className="hg-price">
        <span className="hg-price__icon">$</span>
        <span className="hg-price__dollars">{dollars}</span>
        <span className="hg-price__cents">{cents}</span>
        <span>/mo</span>
      </span>
    );

  }
}

SimplePrice.propTypes = {
  price: PropTypes.number
};

SimplePrice.defaultProps = {
  price: 0
};

export default SimplePrice;
