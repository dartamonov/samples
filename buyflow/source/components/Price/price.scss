.hg-price {
  margin-bottom: 0;
  white-space: nowrap;
}
