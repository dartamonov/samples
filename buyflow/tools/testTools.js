import renderer from 'react-test-renderer';

export const takeSnapshot = x => renderer.create(x).toJSON();