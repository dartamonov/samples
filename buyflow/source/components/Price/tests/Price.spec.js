import React from 'react';
import Price from '../Price.jsx';
import {takeSnapshot} from '../../../tools/testTools';

describe('Price component', () => {
  it('handles normal price', () => {
    const snapshot = takeSnapshot(
      <Price price={9.45} />
    );

    expect(snapshot).toMatchSnapshot();
  });
});
