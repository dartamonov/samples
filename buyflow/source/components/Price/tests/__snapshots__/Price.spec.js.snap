// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price component handles normal price 1`] = `
<span
  className="hg-price"
>
  <span
    className="hg-price__icon"
  >
    $
  </span>
  <span
    className="hg-price__dollars"
  >
    9
  </span>
  <span
    className="hg-price__cents"
  >
    .45
  </span>
  <span>
    /mo
  </span>
</span>
`;
